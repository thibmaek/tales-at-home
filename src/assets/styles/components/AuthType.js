 import { StyleSheet } from 'react-native';
 import { white, accentBlue, fontIsTitle, fontIsLabel } from 'src/assets/styles/vars';

 export default StyleSheet.create({
   container: {
     flexDirection: `column`,
     flex: 1,
   },
   loginContainer: {
     flexDirection: `column`, alignItems: `center`,
     flex: 1,
   },
   loginTopContainer: {
     flexDirection: `column`,
     alignItems: `center`,
     justifyContent: `center`,
     flex: 1,
   },
   title: {
     marginBottom: 75,
     ...fontIsTitle, fontSize: 50, fontWeight: `bold`,
     color: white,
     backgroundColor: `transparent`,
   },
   textInput: {
     width: 280,
     marginBottom: 20,
     borderBottomColor: white, borderBottomWidth: 1, borderStyle: `solid`,
   },
   input: {
     height: 30,
     color: white,
     marginTop: 30,
   },
   button: {
     width: 280,
     backgroundColor: white,
     borderWidth: 0, borderRadius: 50,
   },
   buttonText: {
     fontWeight: `bold`,
     color: accentBlue,
   },
   anonLogin: {
     marginBottom: 50,
     borderBottomColor: white,
     borderBottomWidth: 0.5,
     borderStyle: `solid`,
     backgroundColor: `transparent`,
   },
   underlineLink: {
     ...fontIsLabel, fontWeight: `normal`, fontSize: 11,
     color: white,
   },
 });
