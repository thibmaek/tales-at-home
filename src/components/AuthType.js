import React, { Component, PropTypes } from 'react';
import { View, Text, TextInput } from 'react-native';
import { Actions } from 'react-native-router-flux';
import LinearGradient from 'react-native-linear-gradient';
import Button from 'apsl-react-native-button';

import { Auth } from 'src/config/firebase';
import capString from 'src/lib/capitalizeString';

import { white } from 'src/assets/styles/vars';
import s from 'src/assets/styles/components/AuthType';

const AUTH_LOGIN = { action: `inloggen`, title: `aanmelden`, method: `login` };

export default class AuthType extends Component {
  constructor(props, context) {
    super(props, context);
    this.state = {};
  }

  _handleSubmit(type) {
    const { email, password } = this.state;
    switch (type) {
    case `login`:
      return (
        Auth.signInWithEmailAndPassword(email, password)
          .catch(e => console.error(e.message))
      );
    case `anon`:
      return (
        Auth.signInAnonymously()
          .catch(e => console.error(e.message))
      );
    default:
      return (
          Auth.createUserWithEmailAndPassword(email, password)
            .catch(e => console.error(e.message))
      );
    }
  }

  render() {
    return (
      <LinearGradient colors={[`#008EFF`, `#00C8FF`]}  style={s.container}>
        <View style={s.loginContainer}>
          <View style={s.loginTitleContainer}>
            <Text style={s.title}>{ this.props.title.toUpperCase() }</Text>
          </View>
          <View style={s.loginFormContainer}>
            <View style={s.inputContainer}>
              <TextInput
                style={s.input}
                placeholder='Wat is je e-mail adres?'
                keyboardType='email-address'
                returnKeyType='next'
                placeholderTextColor={white}
                onChangeText={email => this.setState({ email })}
              />
            </View>
            <View style={s.inputContainer}>
              <TextInput
                style={s.input}
                secureTextEntry={true}
                placeholder='Wat is je wachtwoord?'
                placeholderTextColor={white}
                onChangeText={password => this.setState({ password })}
              />
            </View>
            <View>
              {this.props.action === `inloggen` ?
                <Button
                  style={s.button}
                  textStyle={s.buttonText}
                  onPress={() => this._handleSubmit(`login`)}
                >
                  { capString(this.props.action) }
                </Button>
                :
                <Button
                  style={s.button}
                  textStyle={s.buttonText}
                  onPress={() => this._handleSubmit(`register`)}
                >
                  { capString(this.props.action) }
                </Button>
              }

            </View>
            {this.props.action === `registreren` ?
              <Text style={s.textLink}
                onPress={() => Actions.setupScene({ authType: AUTH_LOGIN })}>
                Ik heb al een account
              </Text>
            : null }

          </View>
          <View style={s.anonLogin}>
            <Text style={s.underlineLink} onPress={() => this._handleSubmit(`anon`)}>
              Doorgaan zonder account
            </Text>
          </View>
        </View>
      </LinearGradient>
    );
  }

  static propTypes = {
    title: PropTypes.string.isRequired,
    action: PropTypes.string.isRequired,
    method: PropTypes.string,
  }
}
