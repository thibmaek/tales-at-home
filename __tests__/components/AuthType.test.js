import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import AuthType from 'src/components/AuthType';

jest.mock(`apsl-react-native-button`, () => `Button`);
jest.mock(`react-native-linear-gradient`, () => `LinearGradient`);

describe(`Component: <AuthType>`, () => {

  it(`should render without children passed`, () => {
    const tree = renderer.create(<AuthType />).toJSON();
    expect(tree).toMatchSnapshot(`AuthType`);
  });

  it(`should render wrapping, without children passed`, () => {
    const tree = renderer.create(<AuthType></AuthType>).toJSON();
    expect(tree).toMatchSnapshot(`AuthType: Wrapping`);
  });

});

describe(`Component: <AuthType {...props}>`, () => {

  it(`should render with a title prop`, () => {
    const tree = renderer.create(<AuthType title='A snapshot title' />).toJSON();
    expect(tree).toMatchSnapshot(`AuthType: <Prop:title>`);
  });

  it(`should render with an action prop`, () => {
    const tree = renderer.create(<AuthType action='Bewerken' />).toJSON();
    expect(tree).toMatchSnapshot(`AuthType: <Prop:title>`);
  });

});
