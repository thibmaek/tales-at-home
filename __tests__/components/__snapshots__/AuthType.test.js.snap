exports[`AuthType 1`] = `
<LinearGradient
  colors={
    Array [
      "#008EFF",
      "#00C8FF",
    ]
  }
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
      }
    }>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "rgba(255, 255, 255, 1)",
            "fontFamily": "Bariol",
            "fontSize": 50,
            "fontWeight": "bold",
            "marginBottom": 75,
            "textAlign": "center",
          }
        } />
      <View
        style={
          Object {
            "borderBottomColor": "rgba(255, 255, 255, 1)",
            "borderBottomWidth": 1,
            "borderStyle": "solid",
            "marginBottom": 20,
            "width": 280,
          }
        }>
        <TextInput
          placeholder="Wat is je e-mail adres?"
          placeholderTextColor="rgba(255, 255, 255, 1)"
          style={
            Object {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "Bariol",
              "fontSize": 16,
              "fontWeight": "normal",
              "height": 30,
              "marginTop": 30,
            }
          } />
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "rgba(255, 255, 255, 1)",
            "borderBottomWidth": 1,
            "borderStyle": "solid",
            "marginBottom": 20,
            "width": 280,
          }
        }>
        <TextInput
          placeholder="Wat is je wachtwoord?"
          placeholderTextColor="rgba(255, 255, 255, 1)"
          secureTextEntry={true}
          style={
            Object {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "Bariol",
              "fontSize": 16,
              "fontWeight": "normal",
              "height": 30,
              "marginTop": 30,
            }
          } />
      </View>
      <View
        style={undefined}>
        <Button
          style={
            Object {
              "backgroundColor": "rgba(255, 255, 255, 1)",
              "borderRadius": 50,
              "borderWidth": 0,
              "width": 280,
            }
          }
          textStyle={
            Object {
              "color": "rgba(0, 122, 255, 1)",
              "fontFamily": "Bariol",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          } />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "borderBottomColor": "rgba(255, 255, 255, 1)",
          "borderBottomWidth": 0.5,
          "borderStyle": "solid",
          "marginBottom": 50,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgba(255, 255, 255, 1)",
            "fontFamily": "Bariol",
            "fontSize": 11,
            "fontWeight": "normal",
          }
        }>
        Inloggen zonder account
      </Text>
    </View>
  </View>
</LinearGradient>
`;

exports[`AuthType: <Prop:title> 1`] = `
<LinearGradient
  colors={
    Array [
      "#008EFF",
      "#00C8FF",
    ]
  }
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
      }
    }>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "rgba(255, 255, 255, 1)",
            "fontFamily": "Bariol",
            "fontSize": 50,
            "fontWeight": "bold",
            "marginBottom": 75,
            "textAlign": "center",
          }
        }>
        A snapshot title
      </Text>
      <View
        style={
          Object {
            "borderBottomColor": "rgba(255, 255, 255, 1)",
            "borderBottomWidth": 1,
            "borderStyle": "solid",
            "marginBottom": 20,
            "width": 280,
          }
        }>
        <TextInput
          placeholder="Wat is je e-mail adres?"
          placeholderTextColor="rgba(255, 255, 255, 1)"
          style={
            Object {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "Bariol",
              "fontSize": 16,
              "fontWeight": "normal",
              "height": 30,
              "marginTop": 30,
            }
          } />
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "rgba(255, 255, 255, 1)",
            "borderBottomWidth": 1,
            "borderStyle": "solid",
            "marginBottom": 20,
            "width": 280,
          }
        }>
        <TextInput
          placeholder="Wat is je wachtwoord?"
          placeholderTextColor="rgba(255, 255, 255, 1)"
          secureTextEntry={true}
          style={
            Object {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "Bariol",
              "fontSize": 16,
              "fontWeight": "normal",
              "height": 30,
              "marginTop": 30,
            }
          } />
      </View>
      <View
        style={undefined}>
        <Button
          style={
            Object {
              "backgroundColor": "rgba(255, 255, 255, 1)",
              "borderRadius": 50,
              "borderWidth": 0,
              "width": 280,
            }
          }
          textStyle={
            Object {
              "color": "rgba(0, 122, 255, 1)",
              "fontFamily": "Bariol",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          } />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "borderBottomColor": "rgba(255, 255, 255, 1)",
          "borderBottomWidth": 0.5,
          "borderStyle": "solid",
          "marginBottom": 50,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgba(255, 255, 255, 1)",
            "fontFamily": "Bariol",
            "fontSize": 11,
            "fontWeight": "normal",
          }
        }>
        Inloggen zonder account
      </Text>
    </View>
  </View>
</LinearGradient>
`;

exports[`AuthType: <Prop:title> 2`] = `
<LinearGradient
  colors={
    Array [
      "#008EFF",
      "#00C8FF",
    ]
  }
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
      }
    }>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "rgba(255, 255, 255, 1)",
            "fontFamily": "Bariol",
            "fontSize": 50,
            "fontWeight": "bold",
            "marginBottom": 75,
            "textAlign": "center",
          }
        } />
      <View
        style={
          Object {
            "borderBottomColor": "rgba(255, 255, 255, 1)",
            "borderBottomWidth": 1,
            "borderStyle": "solid",
            "marginBottom": 20,
            "width": 280,
          }
        }>
        <TextInput
          placeholder="Wat is je e-mail adres?"
          placeholderTextColor="rgba(255, 255, 255, 1)"
          style={
            Object {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "Bariol",
              "fontSize": 16,
              "fontWeight": "normal",
              "height": 30,
              "marginTop": 30,
            }
          } />
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "rgba(255, 255, 255, 1)",
            "borderBottomWidth": 1,
            "borderStyle": "solid",
            "marginBottom": 20,
            "width": 280,
          }
        }>
        <TextInput
          placeholder="Wat is je wachtwoord?"
          placeholderTextColor="rgba(255, 255, 255, 1)"
          secureTextEntry={true}
          style={
            Object {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "Bariol",
              "fontSize": 16,
              "fontWeight": "normal",
              "height": 30,
              "marginTop": 30,
            }
          } />
      </View>
      <View
        style={undefined}>
        <Button
          style={
            Object {
              "backgroundColor": "rgba(255, 255, 255, 1)",
              "borderRadius": 50,
              "borderWidth": 0,
              "width": 280,
            }
          }
          textStyle={
            Object {
              "color": "rgba(0, 122, 255, 1)",
              "fontFamily": "Bariol",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }>
          Bewerken
        </Button>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "borderBottomColor": "rgba(255, 255, 255, 1)",
          "borderBottomWidth": 0.5,
          "borderStyle": "solid",
          "marginBottom": 50,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgba(255, 255, 255, 1)",
            "fontFamily": "Bariol",
            "fontSize": 11,
            "fontWeight": "normal",
          }
        }>
        Inloggen zonder account
      </Text>
    </View>
  </View>
</LinearGradient>
`;

exports[`AuthType: Wrapping 1`] = `
<LinearGradient
  colors={
    Array [
      "#008EFF",
      "#00C8FF",
    ]
  }
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
      }
    }>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "rgba(255, 255, 255, 1)",
            "fontFamily": "Bariol",
            "fontSize": 50,
            "fontWeight": "bold",
            "marginBottom": 75,
            "textAlign": "center",
          }
        } />
      <View
        style={
          Object {
            "borderBottomColor": "rgba(255, 255, 255, 1)",
            "borderBottomWidth": 1,
            "borderStyle": "solid",
            "marginBottom": 20,
            "width": 280,
          }
        }>
        <TextInput
          placeholder="Wat is je e-mail adres?"
          placeholderTextColor="rgba(255, 255, 255, 1)"
          style={
            Object {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "Bariol",
              "fontSize": 16,
              "fontWeight": "normal",
              "height": 30,
              "marginTop": 30,
            }
          } />
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "rgba(255, 255, 255, 1)",
            "borderBottomWidth": 1,
            "borderStyle": "solid",
            "marginBottom": 20,
            "width": 280,
          }
        }>
        <TextInput
          placeholder="Wat is je wachtwoord?"
          placeholderTextColor="rgba(255, 255, 255, 1)"
          secureTextEntry={true}
          style={
            Object {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "Bariol",
              "fontSize": 16,
              "fontWeight": "normal",
              "height": 30,
              "marginTop": 30,
            }
          } />
      </View>
      <View
        style={undefined}>
        <Button
          style={
            Object {
              "backgroundColor": "rgba(255, 255, 255, 1)",
              "borderRadius": 50,
              "borderWidth": 0,
              "width": 280,
            }
          }
          textStyle={
            Object {
              "color": "rgba(0, 122, 255, 1)",
              "fontFamily": "Bariol",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          } />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "borderBottomColor": "rgba(255, 255, 255, 1)",
          "borderBottomWidth": 0.5,
          "borderStyle": "solid",
          "marginBottom": 50,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgba(255, 255, 255, 1)",
            "fontFamily": "Bariol",
            "fontSize": 11,
            "fontWeight": "normal",
          }
        }>
        Inloggen zonder account
      </Text>
    </View>
  </View>
</LinearGradient>
`;
